src/rgw/rgw_rest_pubsub_common.cc

```diff
@@ -24,7 +31,7 @@ bool validate_and_update_endpoint_secret(rgw_pubsub_sub_dest& dest, CephContext
   ceph_assert(user.empty() == password.empty());
   if (!user.empty()) {
       dest.stored_secret = true;
-      if (!rgw_transport_is_secure(cct, env)) {
+      if (!verify_security(cct, env)) {
```

Comment:
`verify_security()` calls `rgw_transport_is_secure()` if the flag is not set.
if the flag is set there is no need to check for transport security, because we allow for user/password on non-secure transport

Reply:
there are many aspects of security  this won't disable, e.g., secure authentication

Reply:
this is not disabling anything. only thing it allows is sending "secrets" (e.g. user/password) on non-secure transport

maybe i should name this function differently?

Reply:
that's all I was saying